backend/domain/response.go

```diff
@@ -1,52 +1,19 @@
 package domain
 
-import (
-	"net/http"
-	"time"
-)
-
-type ResponseError struct {
-	// The server error message.
-	Msg string `json:"message"`
-	// The https status error.
-	Sts int `json:"status"`
-	// Time in which the error has occurred.
-	TimeStamp time.Time `json:"timestamp"`
-}
+type ErrType int
 
-func (r ResponseError) Error() string {
-	return r.Msg
-}
-
-func NewResponseError(msg string, sts int) ResponseError {
-	return ResponseError{
-		Msg:       msg,
-		Sts:       sts,
-		TimeStamp: time.Now(),
-	}
-}
-
-func NotFound(msg string) ResponseError {
-	return NewResponseError(msg, http.StatusNotFound)
-}
-
-func InternalServerError(msg string) ResponseError {
-	if msg == "" {
-		msg = "We encountered an internal error while processing the request."
-	}
-	return NewResponseError(msg, http.StatusInternalServerError)
-}
+const (
```

Comment:
hmm, I'd recommend reordering these such that the first one is `Internal`. Reason being: the first value of `iota` is 0, which is the zero-value for your type `ErrType`. Right now your zero value for this type is `BadInput`, which is pretty specific. `Internal` is pretty generic, and also the case of "I meant to specify a value of `ErrType` but forgot to" would be an Internal error.